python/lsst/daf/butler/registries/sqlPreFlight.py

```diff
@@ -222,15 +258,20 @@ def selectDimensions(self, originInfo, expression, neededDatasetTypes, futureDat
                 regionHolders.append(regionHolder)
 
                 # We also have to include regions from each side of the join
-                # into resultset so that we can filter-out non-overlapping
+                # into result set so that we can filter-out non-overlapping
                 # regions.
-                regionColumnIndices[regionHolder.name] = len(selectColumns)
-                regionColumn = self.registry._schema.tables[regionHolder.name].c.region
-                selectColumns.append(regionColumn)
+                if regionHolder.name not in regionColumnIndices:
```

Comment:
Why is this now in a conditional block?

Reply:
To account for the possibility that the same dimension appears multiple times in this loop, as a part of multiple different join tables (e.g. visit-skypix and visit-tract).  Prior to `MeasureMergedCoaddSourcesTask`, we'd only ever had one spatial join table in a preflight query, and that hid this bug.  Will add comment to that effect.